### PR TITLE
STONEINTG-769: Add GitLab commit status updates for integration tests

### PR DESCRIPTION
**Pull request by MartinBasti**
## TODO
- [x] logging
- [x] gitlab client mockking
- [x] test coverage

## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: status/reporter_gitlab.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+func (r *GitLabReporter) Initialize(ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot) error {
+	token, err := GetPACGitProviderToken(ctx, r.k8sClient, snapshot)
+	if err != nil {
+		r.logger.Error(err, "failed to get token from snapshot",
+			"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
+		return fmt.Errorf("failed to get PAC token for gitlab provider: %w", err)
+	}
+
+	annotations := snapshot.GetAnnotations()
+	repo_url, ok := annotations[gitops.PipelineAsCodeRepoURLAnnotation]
```
Comment (dirgim) [inline status/reporter_gitlab.go:73]: suggestion: `repo_url` -> `repoURL`, or maybe even `repoURLAnnotation` for clarity?

**Review by dirgim — APPROVED** [review]
LGTM aside the small naming suggestion. Nice work on refactoring too.

**Review by jsztuka — APPROVED** [review]
lgtm

File: status/reporter_gitlab.go (hunk 1 above)
Comment (MartinBasti) [reply status/reporter_gitlab.go]: renamed

**Pull request merged**